src/sorted_vector.jl

```diff
@@ -0,0 +1,52 @@
+"""
+A `SortedVector` behaves like a standard Julia `Vector`, except that its elements are stored in sorted order, with an optional function `by` that determines the sorting order in the same way as `Base.searchsorted`.
+"""
+immutable SortedVector{T, F<:Function}
+    data::Vector{T}
+    by::F
+
+    function SortedVector(data::Vector{T}, by::F)
+        new(sort(data), by)
```

Comment:
I think the constructor shouldn't call `sort`. You end up needing `SortedVector(v, dont_sort=true)`, instead of simply writing `SortedVector(v)`, `SortedVector(sort(v))`, `SortedVector(sort!(v))` as needed.

To support different orderings, `SortedVector` should have a `Base.Order.Ordering` type parameter.